manifest/services.py

```diff
@@ -280,8 +312,32 @@ def persist(self):
         """
         self.make_paths_relative()
         try:
-            with open(self.path_manifest, 'w') as fmanifest:
+            with open(self.path_manifest, "w") as fmanifest:
                 fmanifest.write(jsonpickle.encode(self.manifest, make_refs=False))
         except EnvironmentError as e:
             self._logger.error(f"COULD NOT write manifest file '{self.path_manifest}'")
-        self._logger.info(f"WROTE manifest file '{self.path_manifest}', session '{self.manifest.session}'")
+        self._logger.info(
+            f"WROTE manifest file '{self.path_manifest}', session '{self.manifest.session}'"
+        )
+
+    def update_manifest_after_run(self) -> None:
+        """Update the manifest document after all the steps have run.
+        Evaluate the statuses based on the manifest steps statuses.
+        """
+        # Reset statuses
+        self._reset_manifest_document_statuses()
+        # Evaluate statuses
+        self.manifest.msg_completion = ManifestStatus.NOT_SET
```

Comment:
This operation is already performed by the '_reset_manifest_document_statuses()' call